python-test-samples/step-functions-local/README.md

```diff
@@ -0,0 +1,185 @@
+
+```json
+"CheckIdentityLambdaMockedSuccess": {
+  "0": {
+    "Return": {
+      "StatusCode": 200,
+      "Payload": {
+        "statusCode": 200,
+        "body": "{\"approved\": true}"
+      }
+    }
+  }
+}
+```
+
+To mock the DetectSentiment retry behavior, define failure and successful mock responses that conform to the [Comprehend.DetectSentiment](https://docs.aws.amazon.com/comprehend/latest/dg/API_DetectSentiment.html#API_DetectSentiment_ResponseElements) API call. Associate the failure mocks to three request attempts, and associate the successful mock to the fourth attempt:
+
+```json
+"DetectSentimentRetryOnErrorWithSuccess": {
+  "0-2": {
+    "Throw": {
+      "Error": "InternalServerException",
+      "Cause": "Server Exception while calling DetectSentiment API in Comprehend Service"
+    }
+  },
+  "3": {
+    "Return": {
+      "Sentiment": "POSITIVE",
+      "SentimentScore": {
+        "Mixed": 0.00012647535,
+        "Negative": 0.00008031699,
+        "Neutral": 0.0051454515,
+        "Positive": 0.9946478
+      }
+    }
+  }
+}
+
+```
+
+Note that Step Functions Local does not validate the structure of the mocked responses. Ensure that your mocked responses conform to actual responses before testing. To review the structure of service responses, either perform the actual service calls using Step Functions or view the documentation for those services.
+
+Next, associate the mocked responses to a test case identifier:
+
+```json
+"RetryOnServiceExceptionTest": {
+  "Check Identity": "CheckIdentityLambdaMockedSuccess",
+  "Check Address": "CheckAddressLambdaMockedSuccess",
+  "DetectSentiment": "DetectSentimentRetryOnErrorWithSuccess",
+  "Add to FollowUp": "AddToFollowUpSuccess",
+  "CustomerAddedToFollowup": "CustomerAddedToFollowupSuccess"
+}
+```
+
+With the test case and mock responses configured, you can use them for testing with Step Functions Local.
+
+### Key Files in the Project
+
+  - [local_testing.asl.json](statemachine/local_testing.asl.json) - State machine definition in ASL
+  - [valid_input.json](statemachine/test/valid_input.json) - State machine unit test input event
+  - [MockConfigFile.json](statemachine/test/MockConfigFile.json) - Unit test mocked responses
+  - [test_step_functions_local.py](tests/unit/src/test_step_functions_local.py) - Unit tests definition
+
+[Top](#contents)
+
+---
+
+## Unit Test
+
+### Prerequisites
+ - Python 3.9 or newer
+ - [Docker](https://www.docker.com/)
+ 
+### Unit Test description
+
+This example contains a [sample event](statemachine/test/valid_input.json) with new user registration data to be processed by the state machine. The unit tests will check how the state machine behaves for each of the following scenarios defined in the [MockConfigFile.json](statemachine/test/MockConfigFile.json) file:
+
```

Comment:
One recommendation is to briefly outline the test flow - reading the state machine, creating the statemachine in local docket, running the below tests, cleanup.